src/mono/mono.proj

```diff
@@ -358,6 +358,16 @@ JS_ENGINES = [NODE_JS]
     </PropertyGroup>
     <Error Text="Expected version: %(_ExpectedVersionLines.Identity) and actual version: %(_ActualVersionLines.Identity) of WASI SDK does not match. Please delete $(WASI_SDK_PATH) folder to provision a new version."
       Condition="'$(ActualWasiSdkVersion)' != '$(ExpectedWasiSdkVersion)'" />
+
+    <!-- LLVM in WASI SDK 22 will call wasm-opt when found on the PATH. But it will fail because wasm-opt can't read the WASM components.
+    After we upgrade to WASI SDK 23, we could use no-wasm-opt LLVM option to avoid this issue.
```

Comment:
Open an issue about this? Possibly a generic "tasks to do when upgrading to WASI SDK 23". It would include removing `wasi-sdk-p2.cmake`.

Reply:
https://github.com/dotnet/runtime/issues/104773